base/pcre.jl

```diff
@@ -24,28 +24,46 @@ function create_match_context()
     return ctx
 end
 
-const THREAD_MATCH_CONTEXTS = Ptr{Cvoid}[C_NULL]
+THREAD_MATCH_CONTEXTS::Vector{Ptr{Cvoid}} = [C_NULL]
 
 PCRE_COMPILE_LOCK = nothing
 
-_tid() = Int(ccall(:jl_threadid, Int16, ())+1)
+_tid() = Int(ccall(:jl_threadid, Int16, ())) + 1
 _nth() = Int(unsafe_load(cglobal(:jl_n_threads, Cint)))
 
 function get_local_match_context()
     tid = _tid()
-    ctx = @inbounds THREAD_MATCH_CONTEXTS[tid]
+    ctxs = THREAD_MATCH_CONTEXTS
```

Comment:
I find it scary that we rely on "implicit" atomic ordering of global for code like this. It looks so innocent and the nuance can get lost during refactoring (even though I understand it is unlikely to happen for this part of code). Rather, I think it'd be easier to encode the atomic ordering by doing something like `const THREAD_MATCH_CONTEXTS = Threads.Atomic(Ptr{Cvoid}[C_NULL])`.

Reply:
Yeah, I am just waiting on https://github.com/JuliaLang/julia/pull/44231 to be able to specify it explicitly. For now, I am okay with relying on the specified behavior implicitly.